src/taipy/core/_repository/_v2/_filesystem_repository.py

```diff
@@ -0,0 +1,135 @@
+from src.taipy.core.common.typing import Entity, Json, ModelType
+from taipy.config.config import Config
+
+from ...exceptions import InvalidExportPath, ModelNotFound
+from .._v2._abstract_repository import _AbstractRepository, _Decoder, _Encoder
+
+
+class _FileSystemRepository(_AbstractRepository[ModelType, Entity]):
+    """
+    Holds common methods to be used and extended when the need for saving
+    dataclasses as JSON files in local storage emerges.
+
+    Some lines have type: ignore because MyPy won't recognize some generic attributes. This
+    should be revised in the future.
+
+    Attributes:
+        model (ModelType): Generic dataclass.
+        dir_name (str): Folder that will hold the files for this dataclass model.
+    """
+
+    def __init__(self, model: Type[ModelType], dir_name: str):
+        self.model = model
+        self._dir_name = dir_name
+
+    @property
+    def dir_path(self):
+        return self._storage_folder / self._dir_name
+
+    @property
+    def _storage_folder(self) -> pathlib.Path:
+        return pathlib.Path(Config.global_config.storage_folder)
+
+    def _save(self, entity: Entity):
+        self.__create_directory_if_not_exists()
+        # TODO: implement _from_entity on models
+        model = self.model._from_entity(entity)
+        self.__get_model_filepath(model.id).write_text(
+            json.dumps(model.to_dict(), ensure_ascii=False, indent=0, cls=_Encoder, check_circular=False)
+        )
+
+    def _get(self, filepath: pathlib.Path) -> Json:
+        with pathlib.Path(filepath).open(encoding="UTF-8") as source:
+            return json.load(source)
+
+    @_retry(Config.global_config.read_entity_retry or 0, (Exception,))
+    def _load(self, model_id: str) -> Entity:
+        try:
+            model = self.model(**self._get(self.__get_model_filepath(model_id)))
+            return model._to_entity()
+        except FileNotFoundError:
+            raise ModelNotFound(str(self.dir_path), model_id)
+
+    @_retry(Config.global_config.read_entity_retry or 0, (Exception,))
+    def _load_all(self, filters: Optional[List[Dict]] = None) -> List[Entity]:
+        entities = []
+        try:
+            for f in self.dir_path.iterdir():
+                if data := self.__filter_by(f, filters):
+                    entities.append(self.model(**data)._to_entity())
+        except FileNotFoundError:
+            pass
+        return entities
+
+    def _delete(self, model_id: str):
+        try:
+            self.__get_model_filepath(model_id).unlink()
+        except FileNotFoundError:
+            raise ModelNotFound(str(self.dir_path), model_id)
+
+    def _delete_all(self):
+        shutil.rmtree(self.dir_path, ignore_errors=True)
+
+    def _delete_many(self, ids: Iterable[str]):
+        for model_id in ids:
+            self._delete(model_id)
+
+    def _search(self, attribute: str, value: Any, filters: List[Dict] = None) -> Optional[Entity]:
```

Comment:
The signature is different from the parent class _AbstractRepository. Is it on purpose?

Reply:
Fixed!